pygmt/xarray_backend.py

```diff
@@ -0,0 +1,72 @@
+"""
+An xarray backend for reading raster grid/image files using the 'gmt' engine.
+"""
+
+import os
+from pathlib import Path
+from typing import Literal
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_list
+from pygmt.src.which import which
+from xarray.backends import BackendEntrypoint
+
+
+class GMTBackendEntrypoint(BackendEntrypoint):
+    """
+    Xarray backend to read raster grid/image files using 'gmt' engine.
+
+    Internally, GMT uses the NetCDF C library to read NetCDF files, and GDAL for GeoTIFF
+    files.
```

Comment:
Need to add a few examples in the docstring.

Reply:
Added a couple of examples in commit 0c6b8a3f225058dbf401daf365b815307760d9ab